src/python/scripts/run_gerp.py

```diff
@@ -28,45 +28,81 @@
-        required=True, description="MSA file (MFA format)"
-    )
-    tree_file = argschema.fields.InputFile(
-        required=True, description="Tree file (Newick format). Must include every species in the MSA."
-    )
-    depth_threshold = argschema.fields.Float(
-        required=False,
-        description="Constrained elements' depth threshold for shallow columns, in substitutions "
-                    "per site. By default, 0.5."
-    )
-    gerp_exe_dir = argschema.fields.InputDir(
-        required=False,
-        description="Path where 'gerpcol' and 'gerpelem' binaries can be found. By default, resort to $PATH."
-    )
+    Returns:
+        None
+    """
+    with open(json_file, 'r') as json_file_handler:
+        align_set = json.load(json_file_handler)
 
+    with open(ce_file) as ce_file_handler:
+        constrained_elements = [ce.split() for ce in ce_file_handler if ce.rstrip() != ""]
 
-if __name__ == "__main__":
-    mod = argschema.ArgSchemaParser(schema_type=InputSchema)
+    # enrich json file with constrained elements info
+    constrained_elems = []
+    for ce in constrained_elements:
+        constrained_elem = {}
+        constrained_elem["start"] = ce[0]
+        constrained_elem["end"] = ce[1]
+        constrained_elem["length"] = ce[2]
+        constrained_elem["score"] = ce[3]
+        constrained_elem["p-val"] = ce[4]
+        constrained_elems.append(constrained_elem)
+    align_set["constraint_elems"] = constrained_elems
+
+    with open(json_file, 'w') as json_file_handler:
+        json.dump(align_set, json_file_handler)
+
+
+def main(param: argparse.Namespace) -> None:
+    ''' Main function of the run_gerp.py script
+
+    This function is running gerpcol that define a gerpscore for every column of the genomic alignment bloc
+    and gerpelem that identify constrained elements across the alignment bloc
+
+    Args:
+        param: argparse.Namespace storing all the script parameters
 
-    cmd = ["gerpcol", "-t", mod.args["tree_file"], "-f", mod.args["msa_file"]]
-    if "gerp_exe_dir" in mod.args:
-        cmd[0] = os.path.join(mod.args['gerp_exe_dir'], cmd[0])
+    Returns:
+        None
+    '''
+
+    cmd = ["gerpcol", "-t", param.tree_file, "-f", param.msa_file]
+    cmd[0] = os.path.join(param.gerp_exe_dir, cmd[0])
     subprocess.run(cmd, check=True)
 
     # By default, gerpcol's ouput filename has the MSA filename plus ".rates" suffix
-    cmd = ["gerpelem", "-f", f"{mod.args['msa_file']}.rates"]
-    if "gerp_exe_dir" in mod.args:
-        cmd[0] = os.path.join(mod.args['gerp_exe_dir'], cmd[0])
-    if "depth_threshold" in mod.args:
-        cmd += ["-d", str(mod.args["depth_threshold"])]
+    cmd = ["gerpelem", "-f", f"{param.msa_file}.rates"]
+    cmd[0] = os.path.join(param.gerp_exe_dir, cmd[0])
+    cmd += ["-d", str(param.depth_threshold)]
     subprocess.run(cmd, check=True)
+
+    add_ce_to_json(f"{param.msa_file}.rates.elems", param.msa_file.replace(".fa", ".json"))
```

Comment:
I advise against using the `str.replace` method to replace the FASTA file extension with a JSON file extension. Using the `str.replace` method in this way would replace any occurrence of the substring `.fa` in the string, and if the MSA file has any other FASTA extension (e.g. `.fasta`, `.mfa`), it will quietly pass back the MSA filename unmodified. 

Another way to approach this would be to replace the FASTA extension using an anchored regex that includes all the FASTA file extensions that might be reasonably expected (e.g. `re.sub("(?<=.)\.(fa|fasta|fna|mfa)$", "", filename)`), though this still won't work if an unfamiliar FASTA extension is used.

A second alternative would be to use the `os.path.splitext` function to remove the file extension. This would have the advantage of not having to know the file extension in advance.

A third option would be to add a `--json_file` argument to the script, so that the user would specify the JSON file explicitly. This puts a slight extra burden on the caller, but it also gives them the freedom to name the JSON file however they like, and it removes a possible source of error from the script.